ui/components/app/currency-input/hooks/useTokenExchangeRate.tsx

```diff
@@ -22,18 +35,37 @@ export default function useTokenExchangeRate(
     shallowEqual,
   );
 
+  const [exchangeRates, setExchangeRates] = useState<Record<string, number>>(
+    {},
+  );
+
   return useMemo(() => {
     if (!tokenAddress) {
       return nativeConversionRate;
     }
 
     const contractExchangeRate =
-      contractExchangeRates[toChecksumAddress(tokenAddress)];
+      contractExchangeRates[tokenAddress] || exchangeRates[tokenAddress];
 
     if (!contractExchangeRate) {
+      fetchTokenExchangeRates(nativeCurrency, [tokenAddress], chainId).then(
+        (addressToExchangeRate) => {
```

Comment:
Is is possible for this to get called too often? Should there be a condition that prevents a new call when one is already loading

Reply:
Great catch! Resolved.